code/game/gamemodes/hivemind/objectives.dm

```diff
@@ -43,6 +43,10 @@
 /datum/objective/hivemind/assimilate
 	explanation_text = "This is a bug. Error:HIVE3"
 
+/datum/objective/hivemind/assimilate/find_target_by_role(role, role_type=0, invert=0)
+	..()
+	return target
```

Comment:
can't you just leave this proc undefined, if you're not using this return value?